data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/AbstractBuffer.java

```diff
@@ -27,7 +27,7 @@
     private final Counter recordsWrittenCounter;
     private final Counter recordsReadCounter;
     private final AtomicLong recordsInFlight;
-    private final AtomicLong recordsInBuffer;
+    protected final AtomicLong recordsInBuffer;
```

Comment:
I have a suggestion below that might make this unnecessary. But, even if we don't go that route, this should be exposed as a `protected` getter method.